authenticate.js

```diff
@@ -0,0 +1,42 @@
+function jsonParse(value, cb) {
+	var err = null, result
+	try {
+		result = JSON.parse(value)
+	} catch (e) {
+		err = e
+	}
+	cb(err, result)
+}
+
+module.exports = function a(emitter, tokenDb, tokenLock) {
+	return function authenticate(token, cb) { //cb(err, credentials)
+		if (!cb) cb = function () {}
+
+		if (!token) {
+			setTimeout(cb, 0, new Error('No token found'))

```

Comment:
I'm guessing you used `setTimeout` instead of `process.nextTick` to be able to pass in the argument without needing an anonymous function?  I do like the simplicity of this call, but `process.nextTick` is more efficient.